src/SectorView.cpp

```diff
@@ -636,64 +649,74 @@ std::vector<SystemPath> SectorView::AutoRoute(const SystemPath &start, const Sys
 	// distance to the start is 0
 	path_dist[0] = 0.f;
 
+	size_t totalSkipped = 0u;
 	while (unvisited.size() > 0) {
 		// find the closest node (for the first loop this will be start)
 		std::vector<SystemPath>::size_type closest_i = *unvisited.begin();
-		for (auto it = unvisited.begin(); it != unvisited.end(); it++) {
+		for (auto it = unvisited.begin(), itEnd = unvisited.end(); it != itEnd; ++it) {
```

Comment:
I don't think that caching end() iterator will actually make any difference. In fact, I expect them to produce identical code by most compilers.
By the way, is there any reason range-based for loop is not used here?
i.e:
` for (auto x : unvisited) { ... } `

Reply:
Also, if instead of iterating over the vector to find the current minimum, heap is used (e.g. std::priority_queue, or it may be easier to use std::set for that [as it can insert/remove/get minimum, all at O(log N), just as heap], the algorithm becomes O(N log N) rather than O(N^2)), see for example here:
https://www.geeksforgeeks.org/dijkstras-shortest-path-algorithm-using-priority_queue-stl/

It requires some rework, so feel free to ignore that comment (or I can rewrite it).

Reply:
This is where I'd actually like to put in a spatial locality search instead of this iterate over everything approach.